modules/service/src/main/scala/lucuma/odb/service/VisitService.scala

```diff
@@ -2,131 +2,121 @@
-     static:        GmosSouth
-   )(using Transaction[F]): F[VisitService.InsertVisitResponse]
-
- }
-
- object VisitService {
-
-   sealed trait InsertVisitResponse extends Product with Serializable
-   object InsertVisitResponse {
-
-     case class NotAuthorized(
-       user: User
-     ) extends InsertVisitResponse
-
-     case class ObservationNotFound(
-       id:         Observation.Id,
-       instrument: Instrument
-     ) extends InsertVisitResponse
-
-     case class Success(
-       vid: Visit.Id
-     )  extends InsertVisitResponse
-
-   }
-
-   def instantiate[F[_]: Concurrent: UUIDGen](using Services[F]): VisitService[F] =
-     new VisitService[F] {
-
-       private def insert(
-         observationId: Observation.Id,
-         instrument:    Instrument,
-         insertStatic:  Option[Visit.Id] => F[Long]
-       ): F[InsertVisitResponse] = {
-         import InsertVisitResponse.*
-
-         val checkUser: Either[NotAuthorized, Unit] =
-           user.role.access match {
-             case Staff | Admin | Service => ().asRight
-             case _                       => /* NotAuthorized(user).asLeft */ ().asRight // for now
-           }
-
-         def insertVisit(v: Visit.Id): F[Either[ObservationNotFound, Unit]] =
-           session
-             .execute(Statements.InsertVisit)(v, observationId, instrument)
-             .as(().asRight)
-             .recover {
-               case SqlState.ForeignKeyViolation(_) =>
-                 InsertVisitResponse.ObservationNotFound(observationId, instrument).asLeft
-             }
-
-         (for {
-           _ <- EitherT.fromEither(checkUser)
-           v <- EitherT.right(UUIDGen[F].randomUUID).map(Visit.Id.fromUuid)
-           _ <- EitherT(insertVisit(v))
-           _ <- EitherT.right(insertStatic(v.some).void)
-         } yield InsertVisitResponse.Success(v)).merge
-       }
-
-       override def insertGmosNorth(
-         observationId: Observation.Id,
-         static:        GmosNorth
-       )(using Transaction[F]): F[InsertVisitResponse] =
-         insert(observationId, Instrument.GmosNorth, gmosSequenceService.insertGmosNorthStatic(observationId, _, static))
-
-       override def insertGmosSouth(
-         observationId: Observation.Id,
-         static:        GmosSouth
-       )(using Transaction[F]): F[InsertVisitResponse] =
-         insert(observationId, Instrument.GmosSouth, gmosSequenceService.insertGmosSouthStatic(observationId, _, static))
-
-     }
-
-   object Statements {
-
-     val InsertVisit: Command[(Visit.Id, Observation.Id, Instrument)] =
-       sql"""
-         INSERT INTO t_visit (
-           c_visit_id,
-           c_observation_id,
-           c_instrument
-         )
-         SELECT
-           $visit_id,
-           $observation_id,
-           $instrument
-       """.command
-
-   }
- }
\ No newline at end of file
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.service
```

Comment:
Sorry for the huge change set here.  The file was inexplicably indented one space to the right.